rescore/artefacts.py

```
@@ -239,51 +196,23 @@ def filesystem_paths_for_finding(
 
 
 def sprint_for_finding(
-    finding: dso.model.ArtefactMetadata,
-    severity: gcm.Severity | None,
-    max_processing_days: gcm.MaxProcessingTimesDays | None,
+    discovery_date: datetime.date,
+    allowed_processing_time: int | None,
     sprints: list[yp.Sprint],
 ) -> yp.Sprint | None:
-    if not severity or not max_processing_days or not sprints:
+    if allowed_processing_time is None or not sprints:
         return None
 
-    max_days = max_processing_days.for_severity(severity=severity)
-    date = finding.discovery_date + datetime.timedelta(days=max_days)
+    date = discovery_date + datetime.timedelta(days=allowed_processing_time)
 
     for sprint in sorted(sprints, key=lambda sprint: sprint.end_date):
         if sprint.end_date.date() > date:
-            break
-    else:
-        logger.warning(f'could not determine target sprint for {finding=} with {severity=}')
-        return None
-
-    return sprint
+            return sprint
 
-
-def _rescorings_and_sprint(
-    artefact_metadatum: dso.model.ArtefactMetadata,
-    rescorings: collections.abc.Iterable[dso.model.ArtefactMetadata],
-    max_processing_days: gcm.MaxProcessingTimesDays | None=None,
-    sprints: list[yp.Sprint]=[],
-) -> tuple[tuple[dso.model.ArtefactMetadata], yp.Sprint]:
-    current_rescorings = rescore.utility.rescorings_for_finding_by_specificity(
-        finding=artefact_metadatum,
-        rescorings=rescorings,
+    logger.warning(
+        f'could not determine target sprint for {discovery_date=} with {allowed_processing_time=}'
```

should this function rather raise (it is okay to have an option telling it not to raise, as we have a a pattern elsewhere). my rationale being to avoid hiding errors (I think warnings might go by unnoticed)

---

I think this should rather be handled within the artefact enumerator, as it takes care of the configured sprint window. Here in this case, the sprint information is purely informational so that there are filter options in the rescoring UI. In case no sprint can be determined for a given finding, this will be displayed in the UI accordingly.